security/authorization/util/util_test.go

```diff
@@ -0,0 +1,112 @@
+			authzArgs:                 map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			// use a policy with == primative and returns a match.
+			expectedEvaluationOutcome: true,
+			expr:                      "connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs:                 map[string]interface{}{"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			// use a policy with startsWith primative and returns no match.
+			expectedEvaluationOutcome: false,
+			expr:                      "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:                 map[string]interface{}{"request.url_path": "/source/pkg.service/test"},
+		},
+		{
+			// use a policy with startsWith and == primatives and returns a match.
+			expectedEvaluationOutcome: true,
+			expr:                      "request.url_path == '/pkg.service/test' && connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs: map[string]interface{}{"request.url_path": "/pkg.service/test",
+				"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			// use a policy that has a field not present in the environment.
+			expectedParsingError: true,
+			expr:                 "request.source_path.startsWith('/pkg.service/test')",
+			authzArgs:            map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			// use an argument that has a filed not included in the environment.
+			expectedEvaluationError: true,
+			expr:                    "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:               map[string]interface{}{"request.source_path": "/pkg.service/test"},
+		},
+	}
+	declarations := []*expr.Decl{decls.NewIdent("request.url_path", decls.String, nil),
+		decls.NewIdent("request.host", decls.String, nil),
+		decls.NewIdent("request.method", decls.String, nil),
+		decls.NewIdent("request.headers", decls.NewMapType(decls.String, decls.String), nil),
+		decls.NewIdent("source.address", decls.String, nil),
+		decls.NewIdent("source.port", decls.Int, nil),
+		decls.NewIdent("source.principal", decls.String, nil),
+		decls.NewIdent("destination.address", decls.String, nil),
+		decls.NewIdent("destination.port", decls.Int, nil),
+		decls.NewIdent("connection.uri_san_peer_certificate", decls.String, nil)}
+
+	env, _ := cel.NewEnv(cel.Declarations(declarations...))
+
+	for _, testInfo := range policyTests {
+		expected, expr, authzArgs := testInfo.expectedEvaluationOutcome, testInfo.expr, testInfo.authzArgs
+		checked, err := convertStringToCheckedExpr(expr, declarations)
+		if err != nil {
+			if testInfo.expectedParsingError == true {
+				continue
+			}
+			t.Errorf("Error in conversion %v", err)
+		}
+		ast := cel.CheckedExprToAst(checked)
+		program, err := env.Program(ast)
+		got, _, gotErr := program.Eval(authzArgs)
+		if gotErr != nil && testInfo.expectedEvaluationError == true {
```

Comment:
what if gotErr == nil && testInfo.expectedEvaluationError? this is a potential failing case, and we want to report it, right?

Reply:
Good catch I will add the logic for that case